app/controllers/teacher_interface/uploads_controller.rb

```diff
@@ -25,15 +25,19 @@ def create
     end
 
     def delete
+      @delete_upload_form = DeleteUploadForm.new
     end
 
     def destroy
-      if ActiveModel::Type::Boolean.new.cast(params.dig(:upload, :confirm))
-        @upload.attachment.purge
-        @upload.destroy!
+      confirm = params.dig(:teacher_interface_delete_upload_form, :confirm)
+      upload = load_upload
```

Comment:
This method is already called for this controller method, due to this line at the top of the file:

```rb
before_action :load_upload, only: %i[delete destroy]
```

so it's already accessible as `@upload` which you can then pass in to the form.